syncer/syncer.go

```diff
@@ -675,10 +680,22 @@ func (s *Syncer) Run(ctx context.Context) error {
 	return err
 }
 
-// Close closes the Syncer's net.Listener.
-func (s *Syncer) Close() error {
+// Shutdown closes the Syncer's net.Listener.
+func (s *Syncer) Shutdown(ctx context.Context) error {
 	err := s.l.Close()
-	s.wg.Wait()
+
+	waitChan := make(chan struct{})
+	go func() {
```

Comment:
Similar thing here. This shouldn't really be necessary. If we run into this, we still got deadlock issues.

Reply:
I definitely agree on the deadlock still being an issue, I wasn't aware the stream would automatically unblock if the listener got closed. That aside though, I always do the `waitChan` pattern to avoid a naked `s.wg.Wait` and consider it good practice?